src/Google.Cloud.Functions.Hosting/HostingInternals.cs

```diff
@@ -166,13 +168,20 @@ internal static IServiceCollection AddServicesForFunctionTarget(IServiceCollecti
             {
                 services
                     .AddScoped<IHttpFunction, CloudEventAdapter>()
+                    .AddSingleton<CloudEventFormatter>(new JsonEventFormatter())
```

Comment:
This always comes before code in startup classes. If multiple classes add a single instance, we'll get the last one that's configured, so that's not too bad.